src/libraries/System.Net.Sockets/src/System/Net/Sockets/DynamicWinsockMethods.cs

```diff
@@ -141,125 +79,27 @@ private IntPtr LoadDynamicFunctionPointer(SafeSocketHandle socketHandle, ref Gui
                 throw new SocketException();
             }
 
-            return ptr;
+            Interlocked.CompareExchange<T?>(ref cache, Marshal.GetDelegateForFunctionPointer<T>(ptr), null!);
```

Comment:
I'm surprised the ! is necessary on the null.  What error do you get without that?

Reply:
Left over from me trying to find the right combination of `!`s and `?`s to make the compiler happy...